### ISSUE TITLE
DNS should use search domains by default?

### ISSUE THREAD

**Issue by ioquatix** [celluloid/celluloid-dns#7]
Following from https://github.com/celluloid/celluloid-io/issues/146#issuecomment-141515313

- If an instance of Celluloid::DNS::Resolver is created, do we consult `resolve.conf` for every request?
- If a user specifies a search prefix, do we consult `resolve.conf` at all?

**Comment by Asmod4n** [comment]
Ruby has a class which deals which ```nameserver```, ```search``` etc from ```resolve.conf``` and the Windows Registry: https://github.com/ruby/ruby/blob/v2_2_3/lib/resolv.rb#L959

**Comment by Asmod4n** [comment]
On OS X you also have to look inside the /etc/resolver directory: https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man5/resolver.5.html

**Comment by ioquatix** [comment]
Yes, I was thinking that on Mac OS X, the `resolv.rb` code wouldn't work correctly.
